client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HttpConnection.ts

```diff
@@ -162,7 +162,7 @@ export class HttpConnection implements IConnection {
             return url;
         }
 
-        if (typeof window === 'undefined') {
+        if (typeof window === 'undefined' || !window || !window.document) {
```

Comment:
do we need the undefined check

Reply:
It's more idiomatic. I guess `!window` will would work for both `null` and `undefined` but the semantic is a bit different.